src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarLoreListener.java

```diff
@@ -0,0 +1,125 @@
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.block.BlockDropItemEvent;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.inventory.PrepareItemCraftEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.persistence.PersistentDataHolder;
+
+import java.util.Arrays;
+
+/**
+ *
+ * @author lexiccn
+ *
+ */
+public class SiegeWarLoreListener implements Listener {
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onSiegeStart(SiegeWarStartEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        //Set name, type, attacker, defender, start time
+        SiegeWarLoreUtil.bannerSiegeStart(event.getSiege());
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onSiegeEnd(SiegeEndEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        //Set outcome, winner, points, end time
+        SiegeWarLoreUtil.bannerSiegeEnd(event.getSiege());
+    }
+
+    @EventHandler
+    public void onInteractBanner(PlayerInteractEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+        if (event.getHand() != EquipmentSlot.HAND) return;
+
+        PersistentDataHolder holder;
+
+        if (event.getClickedBlock() != null && event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+            if (!SiegeWarLoreUtil.isLoreItem(event.getClickedBlock().getState(), "siege_banner")) return;
+            holder = (PersistentDataHolder) event.getClickedBlock().getState();
+        } else if (event.getItem() != null && event.getAction() == Action.RIGHT_CLICK_AIR) {
+            if (!SiegeWarLoreUtil.isLoreItem(event.getItem(), "siege_banner")) return;
+            holder = event.getItem().getItemMeta();
+        } else return;
+
+        SiegeWarLoreUtil.sendBannerChat(holder, event.getPlayer());
+    }
+
+    @EventHandler
+    public void onPrepareCraft(PrepareItemCraftEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        ItemStack resultItem = event.getInventory().getResult();
+        if (resultItem == null) return;
+        if (resultItem.getType() != Material.SHIELD) return;
+
+        ItemStack bannerItem = Arrays.stream(event.getInventory().getMatrix())
+                .filter(itemStack -> SiegeWarLoreUtil.isLoreItem(itemStack, "siege_banner"))
+                .findFirst().orElse(null);
+        if (bannerItem == null) return;
+
+        SiegeWarLoreUtil.setShieldStackFromHolder(resultItem, bannerItem.getItemMeta());
+    }
+
+    @EventHandler(ignoreCancelled = true)
+    public void onTryStartSiege(PreSiegeCampEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+        if (SiegeWarLoreUtil.isLoreItem(event.getFlag().getState(), "siege_banner")) return;
```

Comment:
This should be `!SiegeWarLoreUtil.isLoreItem(event.getFlag().getState(), "siege_banner")` no?

We only want to cancel the event if its a old banner being used, no?

Reply:
indeed... not sure how this one got flipped cus i remember testing it successfully, will have to retest it 🤔

Reply:
Tracked it down to event.getBlockState() in onDropBanner being changed to event.getBlock().getState(), mistake going back to redoing the Util to use BlockState/ItemStack

Retested everything (Siege start/end, try to start with lore banner, craft shield, interact with banner block/item, place/break lore banner again, break siege banner) and it all works after flipping that back and updating this 👍